lightning/cogs/automod/ui.py

```diff
@@ -191,3 +203,338 @@ async def warn_ban(self, itx: discord.Interaction, button: discord.ui.Button):
         self.choice = "warn_ban"
         await itx.response.edit_message(view=None)
         self.stop()
+
+
+class FakeCtx:
+    def __init__(self, member) -> None:
+        self.author = member
+
+
+class GatekeeperRoleView(BasicMenuLikeView):
+    @discord.ui.select(cls=discord.ui.RoleSelect, placeholder="Select a role", min_values=1, max_values=1)
+    async def callback(self, itx: discord.Interaction[LightningBot], select: discord.ui.RoleSelect):
+        assert itx.guild is not None
+
+        role = select.values[0]
+
+        if role >= itx.guild.me.top_role:
+            await itx.response.send_message("You cannot use this role because it is higher than my role!",
+                                            ephemeral=True)
+            return
+
+        if has_dangerous_permissions(role.permissions):
+            await itx.response.send_message("You cannot use this role because it contains permissions that are deemed"
+                                            " dangerous!", ephemeral=True)
+            return
+
+        await self.insert_role(itx, role)
+        await itx.response.send_message(f"Set {role.name} ({role.mention}) as the gatekeeper role!")
+        self.stop(interaction=itx)
+
+    async def insert_role(self, interaction: discord.Interaction[LightningBot], role: discord.Role):
+        query = """INSERT INTO guild_gatekeeper_config (guild_id, role_id)
+                   VALUES ($1, $2)
+                   ON CONFLICT (guild_id)
+                   DO UPDATE SET role_id=EXCLUDED.role_id;"""
+        await interaction.client.pool.execute(query, role.guild.id, role.id)
+
+    @staticmethod
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (docstrings):** Please update the docstring for function: `create_permission_overwrites`

Reason for update: The function now also returns the number of skipped channels, in addition to successes and failures.

Suggested new docstring:
```
Creates permission overwrites for the gatekeeper role in all channels. Returns the number of successful overwrites, failures, and skipped channels due to lack of permissions.
```